### PR TITLE
add devenv to simplify local dev and ci

### PR DESCRIPTION
**Pull request by sebastianwebber**
TL;DR;  this PR uses a tool called https://devenv.sh that adds support to run the ci tasks locally.
## Changes

What it does:
- [x] builds the go app with nix
- [x] builds the go with gomod2nix
- [x] add pre-commits checks like:
  - [x] `govet`
  - [x] `golangci-lint`
  - [x] `go tests`
- [x] have all ci actions run though devenv

What is missing:
- [x] fix the vpc error that is not related
- [x] update the branch protection rules to reflect the new job names

what is optional but cool:
- [x] build docker images using same binary built
- [ ] build multi-arch images for `x86_64` and `aarch64`
- [ ] push images to the registry

## Usage

> [!IMPORTANT]
> The pre-commit hooks will prevent committing if they fail.
>  image

Start by installing devenv: https://devenv.sh/getting-started/

to build the packages:

```shell
devenv build
```

to run tests and all pre-commit checks:

```shell
devenv test
````

to run a dev environment with all tools installed and configured:

```shell
devenv shell
```

TODO:
- [ ] update readme with details about how to use it and install it at the read me
- [ ] remove docker image that is not useful/necessary

### CODE REVIEW

**Comment by Khyme** [comment]
Sounds great 🤩

File: image.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{ pkgs, app, name, version, ... }:
+pkgs.dockerTools.buildImage {
```
Comment (minkimipt) [inline image.nix:2]: how is docker used in this repo? We are not building any containers here. Is it required here?

**Review by minkimipt — COMMENTED** [review]
I asked a few questions and the conflict in the main.go need to be resolved. Do you think we should do that while we are doing the release with all those fixes? It doesn't look that CI change is going to help with any of those. What do you think of making a release and changing the CI afterwards independently?

File: image.nix (hunk 1 above)
Comment (sebastianwebber) [reply image.nix:2]: it is not, just leaving as example. we may remove if you think that is a problem.
